src/ui/PrimitiveList.tsx

```diff
@@ -93,7 +113,7 @@ export class PrimitiveGroup extends Component {
         </div>
         {expanded ?
           <PrimitiveList
-            primitives={group.primitives}
+            primitives={group.primitives as LanguagePrimitive[]}
```

Comment:
We use `PrimitiveGroup` in ways that are not typesafe. In certain cases, a `PrimitiveGroup` is "flat" and contains only `Primitive` objects. In other cases it's a tree that contains both `Primitive` objects and other `PrimitiveGroup`s. But we don't have a way of specifying whether we are using a flat one or a tree one, so I had to resort to casting in a few places.

Reply:
Ah, good point. How ugly is casting, in TypeScriptland? Would it be preferable to define an iterator on PrimitiveGroup that converts any of them into an array of LanguagePrimitives?

Reply:
It's not necessarily ugly, but an iterator would be better.

Reply:
Is this something you can add, if you feel it's worthwhile?

Reply:
yeah, I'll try and get to it tomorrow.

Reply:
Ok, this is done now.